### DISCUSSION TITLE
Time to take the first photo

### DISCUSSION THREAD

**Discussion by cstone99** [raspberrypi/picamera2#704] [Q&A] [answered]
Hello,

While timing my code I've noticed it takes significantly longer to take the first photo after starting the camera than those following it. I'm assuming this has to do with some type of configuration or control setting but it would be great to know why. Specifically is there a way to reduce the time for the first photo, or is that unavoidable?

Thanks

**Comment by davidplowman** [comment, marked as answer]
There are 2 things going on.

Firstly, it takes a while for the camera to start and for the first frame to come out. If you stop the camera and then restart it to capture a picture, you will always have to wait this long.

Secondly, the camera system normally drops several frames the very first time it starts. It does this to give things like AEC/AGC and AWB some time to settle. This only happens the very first time you start the camera after having "opened" it. You can stop the camera and then restart it, and it won't drop these frames.

This second delay, waiting for those algorithms to settle, can be avoided if you specify your own fixed values for the exposure time and analogue gain, and for the white balance colour gains.
